testutil/integration/integration_setup.go

```diff
@@ -0,0 +1,97 @@
+package integration
+
+import (
+	"fmt"
+	"testing"
+
+	errorsmod "cosmossdk.io/errors"
+	"cosmossdk.io/log"
+	storetypes "cosmossdk.io/store/types"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	dbm "github.com/cosmos/cosmos-db"
+	"gotest.tools/v3/assert"
+
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+)
+
+type IntegrationTestApp struct {
+	*baseapp.BaseApp
+	t                  *testing.T
+	InterfaceRegistry  codectypes.InterfaceRegistry
+	Ctx                sdk.Context
+	QueryServiceHelper *baseapp.QueryServiceTestHelper
```

Comment:
The reason I had made them private and added getter was so that you cannot not overwrite it by mistake.
Do you have a use case for having it public?

Reply:
not really, this setup needs code refactoring. That wasn't done yet.